exec.go

```diff
@@ -53,72 +65,94 @@ var execCommand = cli.Command{
 	},
 }
 
-func execProcess(context *cli.Context, args []string) (int, error) {
+func execProcess(context *cli.Context) (int, error) {
 	container, err := getContainer(context)
 	if err != nil {
 		return -1, err
 	}
-
-	bundle := container.Config().Rootfs
-	if err := os.Chdir(path.Dir(bundle)); err != nil {
+	var (
+		detach = context.Bool("detach")
+		rootfs = container.Config().Rootfs
+	)
+	rootuid, err := container.Config().HostUID()
+	if err != nil {
 		return -1, err
 	}
-	spec, _, err := loadSpec(specConfig, runtimeConfig)
+	p, err := getProcess(context, path.Dir(rootfs))
+	if err != nil {
+		return -1, err
+	}
+	process := newProcess(*p)
+	tty, err := setupIO(process, rootuid, context.String("console"), p.Terminal, detach)
 	if err != nil {
 		return -1, err
 	}
+	if err := container.Start(process); err != nil {
+		return -1, err
+	}
+	if pidFile := context.String("pid-file"); pidFile != "" {
+		if err := createPidFile(pidFile, process); err != nil {
+			return -1, err
```

Comment:
Shouldn't we kill a process in case of failure? Because it seems for me like broken flow.